component/registry.go

```diff
@@ -28,9 +29,40 @@ var (
 	parsedNames = map[string]parsedName{}
 )
 
+// ModuleSystem is a mechanism responsible for allowing components to create other components, or
+// flow systems.
+type ModuleSystem interface {
+	// NewModuleDelegate creates a new, un-started ModuleDelegate.
+	NewModuleDelegate(id string) ModuleDelegate
+}
+
+// ModuleDelegate is a controller for running components within a Module.
+type ModuleDelegate interface {
+	// LoadConfig parses River config and loads it into the ModuleDelegate.
+	// LoadConfig can be called multiple times, and called prior to
+	// [ModuleDelegate.Run].
+	LoadConfig(config []byte, args map[string]any, onExport Export) error
+
+	// Run starts the ModuleDelegate. No components within the ModuleDelegate
+	// will be run until Run is called.
+	//
+	// Run blocks until the provided context is canceled.
+	Run(context.Context)
+
+	// ComponentHandler returns an HTTP handler which exposes endpoints of
+	// components managed by the ModuleDelegate.
+	ComponentHandler() http.Handler
+}
+
+// Export is used for onExport of the ModuleDelegate
+type Export func(exports map[string]any)
```

Comment:
`Export` as a type name might be a little confusing, since it sounds like it holds data rather than being a function. Can this be called `ExportFunc`, `ExportHandler`, or something along those lines that make it clearer from the name it's a callback?